example/integration_test/cardano_ledger_address_derivation_tests.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:ledger_cardano/ledger_cardano.dart';
+import 'package:ledger_flutter/ledger_flutter.dart';
+
+import 'byron_address_test_cases.dart';
+import 'shelley_address_test_cases.dart';
+import 'test_utils.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('Cardano Ledger App Address Derivation Tests', () {
+    late Ledger ledger;
+    late CardanoLedgerApp cardanoApp;
+    late LedgerDevice device;
+
+    setUpAll(() async {
+      ledger = Ledger(options: LedgerOptions(maxScanDuration: const Duration(seconds: 5)));
+      cardanoApp = CardanoLedgerApp(ledger);
+      device = await ledger.scan().first;
+      print("Connecting to device: ${device.id}");
+      await ledger.connect(device);
+      print("Connected to device: ${device.name}");
+    });
+
+    group('Should successfully derive Byron address', () {
+      for (var testCase in byronTestCases) {
+        test(testCase.testName, () async {
+          final isAppXS = (await cardanoApp.getVersion(device)).flags.isAppXS;
```

Comment:
Any idea what this appXs means? I don't like that the same test changes behaviour based on a ledger device flag. This should instead be split to 2 tests:

a) A test for `isAppXs == true` 
b) another test for `isAppXs == false`

For test (a):
* If the device gives `isAppXs == true`, run the test (a) and check for expected result. 
* If the device gives `isAppXs == false`, log a warning message to console that the test cannot be run because `isAppXs` is not true and continue to the next test without doing anything

For test (b): basically same as above. If isAppXs is true on device, skip the test with warning.

P.s. you can actually split this whole thing into 2 `groups`:
* `Should successfully derive Byron address - isAppXs true`
* `Should successfully derive Byron address - isAppXs false`

And you can skip the entire group instead of skipping test by test